### ISSUE TITLE
Alternate methods of calculating free energies on GROMACS other than nonequilibrium methods

### ISSUE THREAD

**Issue by mrshirts** [openforcefield/fah-alchemy#3]
**In broad terms, what are you trying to do?**

Calculate binding free energies using expanded ensemble simulations, AWH, or potentially simple "array of lambda" simulations.  Hamiltonian replica exchange is probably out of scope (requiring multiple processors, and one can't do that with F@H). 

**How do you believe using this project would help you to do this?**

I think it's important to have alternate methods to calculate free energies to validate results.  Expanded ensembles potentially have different success and failure modes than nonequilibrium switching simulations, because they go back and forth along the lambda collective variable, allowing the same simulation to escape from metastable minima.  I'm not saying they are necessarily better that nonequilibrium switching, but they are different, which is a value in and of itself.   AWH is a very similar approach (almost exactly similar in theory, somewhat different in implementation), though I'm not as familiar with the implementation.  

**What problems do you anticipate with using this project to achieve the above?**

Two logistic issues are that the simulations need to copy information from one simulation to the next in the same chain of simulations, and that the log files and .edr need to be analyzed during/after the simulations to obtain the free energies. 

Note that Vince Voelz is heavily using these calculations for F@H COVID Moonshot simulations and others, and so has workflows that work well that can be borrowed.

**Comment by jchodera** [comment]
I agree it's important to support a variety of strategies! Ultimately, I think single-replica methods like [SAMS](https://www.stat.rutgers.edu/home/ztan/Publication/SAMS_preprint.pdf) and [TSS](https://arxiv.org/abs/2112.05109) will be optimal here (for both absolute and relative free energy calculations).

**Comment by mrshirts** [comment]
I agree we need single replica methods for F@H. Expanded ensemble is the general category of algorithm, and SAMS and AWH are two very similar ways to accumulate the biases. As appears to be TSS, though I need to familiarize myself with the details.

**Comment by dotsdl** [comment]
Raw notes from story review, shared here for visibility:

- expanded ensemble approaches (single simulation REPEX) should be supported, not just non-equilibrium switching; also want plain old array of lambdas if possible
	- in this case, each edge for e.g. RBFE calculations would itself correspond to two sets (solvent, complex) of multiple simulations corresponding to an array of lambdas
	- for ABFE calculations, same story, though a bit different in flavor (transformation is a complete disappearance atoms, not a transmutation)
	- edges should generally map to multiple sets (typically two, solvent and complex) of multiple simulations (at least one, but possibly many, at fixed or variable lambdas)
- in terms of execution, do need to make sure that clean continuation is supported
- clarification question: for AWSH and other expanded-ensemble simulations, do we do the switching in the same simulation, just with a relaxation time compared to NES, or is it done in a chain of separate simulations but with one starting from the end coordinates of the last?
